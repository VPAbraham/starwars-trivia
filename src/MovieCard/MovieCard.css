.h3--movie-card {
  font-size: 1.5rem;
}