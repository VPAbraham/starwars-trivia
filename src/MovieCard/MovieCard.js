import React from 'react';
import { Link } from 'react-router-dom'
import './MovieCard.css';


const MovieCard = ({ title, episodeId, releaseDate, characters, openingCrawl, poster }) => {
  
  return (
    <div className='div div--movie-card'>
      <h3 className='h3 h3--movie-card'>{title}</h3>
      <p>{episodeId} {releaseDate}</p>
      <img className='img img--poster' src={poster} alt='bang'/>
      <Link to='/characters'>
        <button data-characters={characters} data-crawl={openingCrawl}>Characters</button>
      </Link>
    </div>
  )
}


export default MovieCard;