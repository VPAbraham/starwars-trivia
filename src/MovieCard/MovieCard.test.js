import React from 'react';
import { shallow } from 'enzyme';
import MovieCard from './MovieCard';

describe('MovieCard', () => {
  let wrapper
  const getCurrentMovieChars = jest.fn()
  let movie = [{
    title: 'StarWars',
    episodeId: 2,
    releaseDate: 'Yesterday',
    characters: [{ name: 'John' }],
    openingCrawl: 'crawling',
    poster: 'img1',
  }]
  beforeEach(() => {
    wrapper = shallow(
      <MovieCard
        movie={movie}
        getCurrentMovieChars={getCurrentMovieChars}
      />
    )
  })
  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  });

  it('should call getCurrentMovieChars when button is clicked', () => {
    wrapper.find('button').simulate('click')
    expect(getCurrentMovieChars).toHaveBeenCalled()
  })
})