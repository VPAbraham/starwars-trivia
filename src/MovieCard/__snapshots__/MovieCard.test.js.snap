// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieCard should match the snapshot 1`] = `
<div
  className="div div--movie-card"
>
  <h3
    className="h3 h3--movie-card"
  />
  <p
    className="p p--movie-episode"
  >
    Episode 
  </p>
  <p
    className="p p--movie-release"
  />
  <img
    alt="bang"
    className="img img--poster"
  />
  <button
    className="button button--movie-characters"
    onClick={[Function]}
  >
    <Link
      to="/characters"
    >
      Characters
    </Link>
  </button>
</div>
`;
